### PR TITLE
Change phlippe resnet tolerance on torchbench for bf16

### LINKED ISSUE #1799: Torchbench phlippe_resnet bfloat16 training got fail_accuracy
### 🐛 Describe the bug

phlippe_resnet  accuracy failed on both LTS and Rolling driver in 07/02 nightly tests
https://github.com/intel/torch-xpu-ops/actions/runs/16001019449
```
loading model: 0it [00:00, ?it/s]
loading model: 0it [00:11, ?it/s]
xpu  train phlippe_resnet                     
E0701 15:17:51.502000 2472533 site-packages/torch/_dynamo/utils.py:3007] RMSE (res-fp64): 0.00734, (ref-fp64): 0.00047 and shape=torch.Size([]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000, use_larger_multiplier_for_smaller_tensor: 0
fail_accuracy

```

Reproducer
```
# Get pytorch wheel from https://github.com/intel/torch-xpu-ops/actions/runs/16001019449
pip install /path/to/pytorch/wheel
git clone https://github.com/pytorch/benchmark
cd benchmark
pip install -r requirements.txt
python install.py phlippe_resnet 
cd ..
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout 1586521461c8dc642735466fc143b7d366a858d0
python benchmarks/dynamo/torchbench.py --accuracy --bfloat16 -d xpu \
     -n10 --training --only phlippe_resnet --cold-start-latency --backend=inductor
```

### Versions

```
Device: PVC 1100
Driver: Rolling 25.13.33276
IGC: 2.10.11-1133~22.04
L0: 1.21.1.0-1123~22.04
Pytorch: 1586521461c8dc642735466fc143b7d366a858d0
Torch-xpu-ops: 9bf5973052feb696bf5776a549f5958cb64f7e55
```

### PR DESCRIPTION
**Pull request by tbohutyn**
With 2.6.0 cosine_similarity was forced. See https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/common.py#L2227
We hit exception because "Double and complex datatype matmul is not supported in oneDNN", since we started to support in later releases cosine_similarity is set to False, which causes fail.
When I forced it to True on latest pytorch, we pass the benchmark.

I'd argue for adding phlippe_resnet here:
https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/torchbench.yaml#L60

Fixes https://github.com/intel/torch-xpu-ops/issues/1799

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo @chenyang78

### CODE REVIEW

**Comment by guangyey** [comment]
Looks reasonable to me.

**Comment by EikanWang** [comment]
@tbohutyn , acutally, `cosine_similarity` cannot guarantee the accuracy. The motivation to mention `cosine_similarity` is to prove the failure is a numeric issue instead of a real failure, right?

File: benchmarks/dynamo/torchbench.yaml
Comment (EikanWang) [inline benchmarks/dynamo/torchbench.yaml:1]: It is irrelevant to this PR. But we need to refine the tolerance logic a little bit because the tolerance of `higher` differs from `even_higher`.
https://github.com/pytorch/pytorch/blob/8d2a06bf5b032b9eee62100f808a87481e48f9c6/benchmarks/dynamo/torchbench.py#L434-L437

Currently, the tolerance of `higher_bf16_xpu` equals `even_higher` for CUDA.

https://github.com/pytorch/pytorch/blob/8d2a06bf5b032b9eee62100f808a87481e48f9c6/benchmarks/dynamo/torchbench.py#L443-L444

It may not make much sense. If we need an even higher tolerance, we should add a section like `even_higher_bf16_xpu`.
